tools/test-proxy/Azure.Sdk.Tools.TestProxy/Store/GitStore.cs

```diff
@@ -2,36 +2,513 @@
+    }
+
+    /// <summary>
+    /// This class provides an abtraction for dealing with git assets that are stored in an external repository. An "assets.json" within a repo folder is used to inform targeting.
+    /// </summary>
     public class GitStore : IAssetsStore
     {
-        public void Push(string pathToAssetsJson, string contextPath) {
-            var config = ParseConfigurationFile(pathToAssetsJson);
+        private HttpClient httpClient = new HttpClient();
+        public GitProcessHandler GitHandler = new GitProcessHandler();
+        public string DefaultBranch = "main";
+        public string FileName = "assets.json";
+
+        public GitStore() { }
+
+        public GitStore(GitProcessHandler processHandler) {
+            GitHandler = processHandler;
+        }
+
+        #region push, reset, restore, and other asset repo implementations 
+        /// <summary>
+        /// Pushes a set of changed files to the assets repo. Honors configuration of assets.json passed into it.
+        /// </summary>
+        /// <param name="pathToAssetsJson"></param>
+        /// <returns></returns>
+        public async Task Push(string pathToAssetsJson) {
+            var config = await ParseConfigurationFile(pathToAssetsJson);
+            var pendingChanges = DetectPendingChanges(config);
+            var onLatestSHA = true;
+
+            if (pendingChanges.Length > 0)
+            {
+                if(!GitHandler.TryRun(config, $"rev-parse origin/{config.AssetsRepoBranch}", out CommandResult result))
+                {
+                    // if we have a nonzero exit code, only code 128 is acceptable.
+                    if (result.ExitCode != 128)
+                    {
+                        throw GenerateInvokeException(result);
+                    }
+                }
+                else
+                {
+                    var retrievedSHA = result.StdOut.Trim();
+                    if (retrievedSHA != config.SHA)
+                    {
+                        onLatestSHA = false;
+                    }
+                }
+
+                if (onLatestSHA)
+                {
+                    if (!GitHandler.TryRun(config, $"checkout -b {config.AssetsRepoBranch}", out CommandResult checkoutResult))
+                    {
+                        throw GenerateInvokeException(checkoutResult);
+                    }
+
+                    if (!GitHandler.TryRun(config, $"add -A .", out CommandResult addResult))
+                    {
+                        throw GenerateInvokeException(addResult);
+                    }
+
+                    if (!GitHandler.TryRun(config, $"commit -m \"Automatic asset update from test-proxy.\"", out CommandResult commitResult))
+                    {
+                        throw GenerateInvokeException(commitResult);
+                    }
+
+                    if (!GitHandler.TryRun(config, $"push origin {config.AssetsRepoBranch}", out CommandResult pushResult))
```

Comment:
I might be missing this from code elsewhere, but how does this end up working? Anyone can inadvertently push, or just pipeline contexts, or...?

Reply:
Right now it's just shelling out to git. I have permissions, so it pushes. This will probably change. I have no idea idea what that will look like, but I'll be crossing that bridge soon with integration jobs.